src/CommandIsolator.cpp

```diff
@@ -214,17 +228,27 @@ process::Future<::mesos::ResourceStatistics> CommandIsolatorProcess::usage(
 process::Future<Nothing> CommandIsolatorProcess::cleanup(
     const ContainerID& containerId) {
   if (m_cleanupCommand.isNone()) {
+    if (m_infos.contains(containerId)) {
```

Comment:
here I agree to let cleanup occur naturally even if m_infos does not contain info.

Reply:
Maybe if we fail for watch & usage callback, maybe it's better to also fail for cleanup ?

Reply:
cleanup semantic does not really say if prepare has been called (or actually finished successfully). If we fail in cleanup, it means user have a feedback about an error in cleanup instead of having the real error shown.

Reply:
So maybe just a warning ? If we try to pass containerConfig each time to the cleanup callback, we can't silently ignore errors because user will expect his script to get all the information

Reply:
warning is the right option indeed